src/invidious/videos.cr

```diff
@@ -877,42 +877,84 @@ def extract_video_info(video_id : String, proxy_region : String? = nil, context_
         }
   ).try { |a| JSON::Any.new(a) } || JSON::Any.new([] of JSON::Any)
 
-  primary_results = player_response.try &.["contents"]?.try &.["twoColumnWatchNextResults"]?.try &.["results"]?
-    .try &.["results"]?.try &.["contents"]?
-  sentiment_bar = primary_results.try &.as_a.select(&.["videoPrimaryInfoRenderer"]?)[0]?
-    .try &.["videoPrimaryInfoRenderer"]?
-      .try &.["sentimentBar"]?
-        .try &.["sentimentBarRenderer"]?
-          .try &.["tooltip"]?
-            .try &.as_s
-
-  likes, dislikes = sentiment_bar.try &.split(" / ", 2).map &.gsub(/\D/, "").to_i64 || {0_i64, 0_i64}
-  params["likes"] = JSON::Any.new(likes)
-  params["dislikes"] = JSON::Any.new(dislikes)
-
-  params["descriptionHtml"] = JSON::Any.new(primary_results.try &.as_a.select(&.["videoSecondaryInfoRenderer"]?)[0]?
-    .try &.["videoSecondaryInfoRenderer"]?.try &.["description"]?.try &.["runs"]?
-      .try &.as_a.try { |t| content_to_comment_html(t).gsub("\n", "<br/>") } || "<p></p>")
-
-  metadata = primary_results.try &.as_a.select(&.["videoSecondaryInfoRenderer"]?)[0]?
-    .try &.["videoSecondaryInfoRenderer"]?
-      .try &.["metadataRowContainer"]?
-        .try &.["metadataRowContainerRenderer"]?
-          .try &.["rows"]?
-            .try &.as_a
+  # Top level elements
+
+  primary_results = player_response
+    .dig?("contents", "twoColumnWatchNextResults", "results", "results", "contents")
+
+  video_primary_renderer = primary_results
+    .try &.as_a.find(&.["videoPrimaryInfoRenderer"]?)
+      .try &.["videoPrimaryInfoRenderer"]
+
+  video_secondary_renderer = primary_results
+    .try &.as_a.find(&.["videoSecondaryInfoRenderer"]?)
+      .try &.["videoSecondaryInfoRenderer"]
+
+  # Likes/dislikes
+
+  toplevel_buttons = video_primary_renderer
+    .try &.dig?("videoActions", "menuRenderer", "topLevelButtons")
+
+  if toplevel_buttons
+    likes_button = toplevel_buttons.as_a
+      .find(&.["toggleButtonRenderer"]["defaultIcon"]["iconType"].as_s.== "LIKE")
```

Comment:
Weird to see the use of the square brace access when dig is being used everywhere else. Any particular reason for this or could it be switched?

Same goes for the line fetching the dislikes button

Reply:
yeah, you're right, that should use `.dig()`!